torch/testing/__init__.py

```diff
@@ -163,14 +180,14 @@ def _compare_tensors_internal(a: torch.Tensor, b: torch.Tensor, *, rtol, atol, e
 
 # Checks if two scalars are equal(-ish), returning (True, None)
 # when they are and (False, debug_msg) when they are not.
-def _compare_scalars_internal(a, b, *, rtol: float, atol: float, equal_nan: bool) -> _compare_return_type:
+def _compare_scalars_internal(a, b, *, rtol: float, atol: float, equal_nan: Union[str, bool]) -> _compare_return_type:
     def _helper(a, b, s) -> _compare_return_type:
         # Short-circuits on identity
-        if a == b or (equal_nan and a != a and b != b):
+        if a == b or (bool(equal_nan) and a != a and b != b):
```

Comment:
This also needs to assert that equal_nan is a valid value and test if that value is "Relaxed" or True directly (not rely on strings being converted to True)